cpp/tests/io/type_inference_test.cu

```diff
@@ -211,7 +211,7 @@ TEST_F(TypeInference, Timestamp)
   auto d_data           = cudf::make_string_scalar(data);
   auto& d_string_scalar = static_cast<cudf::string_scalar&>(*d_data);
 
-  std::size_t constexpr size = 3;
+  std::size_t constexpr size = 2;
```

Comment:
Should this be set from the `string_offset` or `string_length` vector `size()` instead?
It appears they should be correlated but I'm not sure.

Reply:
Yeah, looks like this variable is not needed, using `string_offset.size()` in `infer_data_type` calls would be less error-prone.

Reply:
Ok. I think that is worth doing here.